libsandbox/canonicalize.c

```diff
@@ -110,13 +110,15 @@ erealpath(const char *name, char *resolved)
 				if (lstat64(rpath, &st))
 					break;
 				if (S_ISLNK(st.st_mode)) {
-					ssize_t cnt = readlink(rpath, rpath, path_max);
+				        char buffer[path_max];
```

Comment:
creates a VLA which is iffy...

Reply:
ON X86_64, the prepocessed compilation stage shows that SB_PATH_MAX (defined as 2*PATH_MAX) expands to  2*4096 or 8192 bytes. Either that, or it is made equal to the returned value of pathconf(path,_PC_PATH_MAX), which is 4096 bytes on ext4, at least.  But it can get worse, as erealpath(.) may call itself recursively, although only once if I interpret the comment to the recursive call therein correctly; yet there seems to be no explicit recursion limit until now. So what to do? Change the function definition and manage a stack of preallocated buffers manually? Write a hull function for readlink(.) using a circular buffer space of sufficient size and depth?

Reply:
what about
```suggestion
				        char buffer[SB_PATH_MAX];
```
? The problem isn't the value of `path_max`, the **real problem is that it's a runtime value** and not a compile time value.

Reply:
In turn the implication would be that code calculating a value for path_max should better be deleted or made part of the configure phase, so that only SB_PATH_MAX remains and translates into a compile time constant. But I suspect that the call to 'pathconf(char *path, int name)'  is there for a reason and that the returned value depends on the possibly remote filesystem determined by the first parameter.  The following snippet is from libsandbox/canonicalize.c, preceding the patched region:
```` 
47 char *
48 erealpath(const char *name, char *resolved)
49 {
           [...]
52         long int path_max;
           [...]
69 #ifdef SB_PATH_MAX
70         path_max = SB_PATH_MAX;
71 #else
72         path_max = pathconf(name, _PC_PATH_MAX);
73         if (path_max <= 0)
74                 path_max = 1024;
75 #endif
           [...]
````